tools/ports/regal.py

```diff
@@ -0,0 +1,157 @@
+                          'regal/RegalFavicon.cpp',
+                          'regal/RegalMac.cpp',
+                          'regal/RegalSo.cpp',
+                          'regal/RegalFilt.cpp',
+                          'regal/RegalXfer.cpp',
+                          'regal/RegalX11.cpp',
+                          'regal/RegalDllMain.cpp',
+                          'md5/src/md5.c',
+                          'jsonsl/jsonsl.c']
+
+            commands = []
+            o_s = []
+
+            for src in srcs_regal:
+                c = os.path.join(dest_path_src, src)
+                o = os.path.join(dest_path_src, src + '.o')
+                shared.safe_ensure_dirs(os.path.dirname(o))
+                commands.append([shared.PYTHON, shared.EMCC, c,
+                                 # specify the defined symbols as the Regal Makefiles does for Emscripten+Release
+                                 # the define logic for other symbols will be handled automatically by Regal headers (SYS_EMSCRIPTEN, SYS_EGL, SYS_ES2, etc.)
+                                 '-DNDEBUG',
+                                 '-DREGAL_NO_PNG=1',
+                                 '-DREGAL_LOG=0',
+                                 '-DREGAL_NO_TLS=1',
+                                 '-DREGAL_THREAD_LOCKING=0',
+                                 '-DREGAL_GLSL_OPTIMIZER=0',
+                                 '-fomit-frame-pointer',
+                                 '-Wno-constant-logical-operand',
+                                 '-fvisibility=hidden',
+                                 '-O2',
+                                 '-o', o,
+                                 '-I' + dest_path_include,
+                                 '-I' + dest_path_regal,
+                                 '-I' + os.path.join(dest_path_md5, 'include'),
+                                 '-I' + dest_path_lookup3,
+                                 '-I' + dest_path_jsonsl,
+                                 '-I' + dest_path_boost,
+                                 '-I' + os.path.join(dest_path_glslopt, 'include'),
+                                 '-I' + os.path.join(dest_path_glslopt,'src','glsl'),
+                                 '-I' + os.path.join(dest_path_glslopt,'src','mesa'),
+                                 '-w'])
+                o_s.append(o)
+
+            ports.run_commands(commands)
+            final = os.path.join(ports.get_build_dir(), 'regal', 'libregal.bc')
+            shared.try_delete(final)
+            shared.Building.link(o_s, final)
+            assert os.path.exists(final)
+            return final
+        return [shared.Cache.get('regal', create, what='port')]
+    else:
+        return []
+
+def process_args(ports, args, settings, shared):
+    if settings.USE_REGAL == 1:
+        get(ports, settings, shared)
+        args += ['-Xclang', '-isystem' + os.path.join(ports.get_build_dir(), 'regal', 'include')]
+    return args
```

Comment:
I would guess that Regal needs full ES2 support - or does it happen to be fine with the WebGL-friendly subset of ES2?

If we need full ES2, we should add
```
    settings.FULL_ES2 = 1
```

Reply:
You are definitively right on this. This solved some issues on projects I am working on!